### PR TITLE
Simplify clientset/kubeconfig handling

### LINKED ISSUE #347: control plane version check shouldn't need authenticator
The error message we get when authenticator binary is missing is pretty obscure:
```
2018-12-14T22:00:10Z [▶]  control plane not ready yet – Get https://7D5055ECFB8FABACA4391973BCC59CB2.sk1.us-east-1.eks.amazonaws.com/version?timeout=32s: exec: exec: "heptio-authenticator-aws": executable file not found in $PATH
```
We should just make this call unauthenticated. In fact, it can be done simply with the HTTP client.

### LINKED ISSUE #520: timed out waiting for control plane "demo-cluster" after 20m0s
Please [search existing issues (open & closed)](https://github.com/weaveworks/eksctl/issues?utf8=%E2%9C%93&q=is%3Aissue) to see if there is a similar one. If there is, add comments or vote where appropriate.

If you did not find a similar issue or if it did not help then ask for help in our [slack channel](https://weave-community.slack.com/messages/eksctl/)

**What help do you need?**
Describe the type of help you need
While creating cluster using eksctl I'm facing below error.

[✔]  all EKS cluster resource for "demo-cluster" had been created
[✔]  saved kubeconfig as "/root/.kube/config"

[✖]  timed out waiting for control plane "demo-cluster" after 20m0s

command used for cluster creation --> **eksctl create cluster  --version=1.11 --name=demo-cluster --nodes=3 --node-type=m4.xlarge --region=us-west-2 --tags owner=demouser**

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

- fix issue with accidential use of authenticator
  - it was causing infinite loops when authenticator not present
  - it was due to too many deep references
- reduce overall code complexity

- Close #347 
- Close #520

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] All integration tests passing (i.e. `make integration-test`)

### CODE REVIEW

**Comment by errordeveloper** [comment]
So one previously could do this:
```Go
clientConfig, err := ctl.NewClientConfig(spec)
// handle err
withAuthenticator := clientConfig.WithExecAuthenticator()
withToken, err := clientConfig.WithEmbeddedToken()
// handle err
clientSet, err := withToken.NewClientSet()
// handle err
```
And that had a side-effect of authenticator `withToken` getting authenticator also. That was due to the fact that `clientcmdapi.Config` uses pointers for everything, and we didn't do deep-copy.

The new version lets do this:
```Go
withAuthenticator, err := ctl.NewClientConfig(spec, false)
// handle err
withToken, err := ctl.NewClientConfig(spec, true)
// handle err
clientSet, err := withToken.NewClientSet()
// handle err
```

Or, as we often just want a clientset with a token, there is a simple shortcut:
```Go
clientSet, err := ctl.NewClientSet(spec) 
```

File: pkg/eks/auth.go [hunk 1]
```diff
@@ -152,84 +151,44 @@ func (c *ClusterProvider) getUsername() string {
...
 
-// NewClientConfig creates a new client config
-// based on "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
-// these are small, so we can copy these, and no need to deal with k/k as dependency
-func (c *ClusterProvider) NewClientConfig(spec *api.ClusterConfig) (*ClientConfig, error) {
-	client, clusterName, contextName := kubeconfig.New(spec, c.getUsername(), "")
-	clientConfig := &ClientConfig{
-		Cluster:     spec,
+// NewClientConfig creates a new client config, if withEmbeddedToken is true
+// it will embed the STS toke, otherwise it will use authenticator exec plugin
```
Comment (palemtnrider) [inline pkg/eks/auth.go]: toke -> token

File: pkg/eks/auth.go [hunk 2]
```diff
@@ -246,12 +205,13 @@ func (c *ClientConfig) NewClientSet() (*clientset.Clientset, error) {
 	return client, nil
 }
 
-// NewClientSetWithEmbeddedToken creates a new API client with an embedded STS token
-func (c *ClientConfig) NewClientSetWithEmbeddedToken() (*clientset.Clientset, error) {
-	clientConfig, err := c.WithEmbeddedToken()
+// NewClientSet creates a new API client in one go with an embedded STS token
+func (c *ClusterProvider) NewClientSet(spec *api.ClusterConfig) (*clientset.Clientset, error) {
```
Comment (palemtnrider) [inline pkg/eks/auth.go]: Code would be easier to read and understand if this was NewClientSetWithStsToken (or similar).  There is a NewClientSet() on ClientConfig and ClusterProvider

Comment (errordeveloper) [reply pkg/eks/auth.go]: True, I just went for shorter name.

Comment (errordeveloper) [reply pkg/eks/auth.go]: I'll make it `NewStdClientSet`.

**Comment by palemtnrider** [comment]
These changes look good to me.  I'm running integration test now.

**Comment by palemtnrider** [comment]
Sorry for the long delay.  Still working on getting a clean integration test run.  I'll try again first thing in the morning. A clean run is all I'm waiting on before approving.

**Pull request merged**